drivers/vulkan/vulkan_context.cpp

```diff
@@ -210,57 +211,55 @@ VkBool32 VulkanContext::_check_layers(uint32_t check_count, const char **check_n
 	return 1;
 }
 
-Error VulkanContext::_create_validation_layers() {
-	VkResult err;
-	const char *instance_validation_layers_alt1[] = { "VK_LAYER_KHRONOS_validation" };
-	const char *instance_validation_layers_alt2[] = { "VK_LAYER_LUNARG_standard_validation" };
-	const char *instance_validation_layers_alt3[] = { "VK_LAYER_GOOGLE_threading", "VK_LAYER_LUNARG_parameter_validation", "VK_LAYER_LUNARG_object_tracker", "VK_LAYER_LUNARG_core_validation", "VK_LAYER_GOOGLE_unique_objects" };
+Error VulkanContext::_get_preferred_validation_layers(uint32_t *count, const char *const **names) {
+	static const std::vector<std::vector<const char *>> instance_validation_layers_alt{
```

Comment:
We have our own vector types, is there a specific reason to use `std::vector` here? Otherwise it would likely be best to replace it with a Godot `Vector`.

Reply:
I wanted the variable `instance_validation_layers_alt` to be `static const`, to avoid recreating it every time we call `_get_preferred_validation_layers()` method. `const` variables need to be initialized upon declaration. It is possible to initialize `std::vector` upon declaration, because `std::vector` implements constructor that takes `std::initializer_list`, so the below code works:
```
static const std::vector<std::vector<...>> instance_validation_layers_alt{
    {vector_0_element_0}, {vector_0_element_1},
    {vector_1_element_0},
    {vector_2_element_0}, etc
}
```

Since `Vector` doesn't have constructor that takes `std::initializer_list`, it means that `instance_validation_layers_alt` can't be either `static` or `const`, and the initialization would look like this:

```
Vector<Vector<...>> instance_validation_layers_alt;
instance_validation_layers_alt.resize(3);
instance_validation_layers_alt[0].push_back(vector_0_element_0);
instance_validation_layers_alt[0].push_back(vector_0_element_1);
instance_validation_layers_alt[1].push_back(vector_1_element_0);
instance_validation_layers_alt[2].push_back(vector_2_element_0);
etc
```

Maybe I'm missing something or maybe there is a completely different and better way of doing this. Pls let me know if you want me to go with `Vector`.

Reply:
That makes sense, thanks!

@hpvb will take a look at implementing this constructor for `Vector` and `LocalVector` as it's indeed something that would be quite useful throughout the codebase.

In the meantime we can likely merge this as is :+1: